Framework/Core/src/CommonDataProcessors.cxx

```diff
@@ -63,6 +68,225 @@ struct InputObject {
 const static std::unordered_map<OutputObjHandlingPolicy, std::string> ROOTfileNames = {{OutputObjHandlingPolicy::AnalysisObject, "AnalysisResults.root"},
                                                                                        {OutputObjHandlingPolicy::QAObject, "QAResults.root"}};
 
+
+void CommonDataProcessors::DataRef2Tree(TFile* f, InputRecord inprec, DataRef ref)
```

Comment:
Why don't you pass the `TTree` rather than the `TFile*`? It would also be nice to have a benchmark with the filling.

Reply:
Actually you could make this a `arrow::Table` -> `TTree` converse.